components/org.wso2.carbon.identity.api.user.session/org.wso2.carbon.identity.api.user.session.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/session/v1/impl/UserIdApiServiceImpl.java

```diff
@@ -63,9 +72,33 @@ public Response terminateSessionBySessionId(String userId, String sessionId) {
     @Override
     public Response terminateSessionsByUserId(String userId) {
 
-        Util.validateUserId(SessionManagementServiceHolder.getRealmService(), userId,
-                ContextLoader.getTenantDomainFromContext());
-        sessionManagementService.terminateSessionsByUserId(userId);
-        return Response.noContent().build();
+        try {
+            UserRealm userRealm = CarbonContext.getThreadLocalCarbonContext().getUserRealm();
+            AbstractUserStoreManager userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager();
+
+            if (userStoreManager == null) {
+                log.debug("Returns Null by UserStore Manager");
+                return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+            }
+
+            String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            String adminUserName = userRealm.getRealmConfiguration().getAdminUserName();
+            String adminUserID = userStoreManager.getUserIDFromUserName(adminUserName);
+
+            if (!StringUtils.equals(username, adminUserName) && StringUtils.equals(userId, adminUserID)) {
+                log.debug("Forbidden operation Collaborator having admin permissions is trying to" +
```

Comment:
check if debug is enabled.

Reply:
Change message as  "Forbidden operation. Admin user is not allowed to terminate the organization owner's sessions"

Reply:
Added in commit 5a50f558230bca28f5468999859a4ece2b25ded1